.github/workflows/docker.yml

```diff
@@ -0,0 +1,35 @@
+name: Docker Image CI
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
```

Comment:
you probably don't want to publish a new artifact a new image on pull requests. Only valid push to main should constitute a new image build

I also like Martin's suggestion to run this at random time occasionally

```yaml
  schedule:
    # Once monthly at a randomly selected time.
    - cron: "24 2 3,18 * *"
```

Reply:
I would like to see more restrictions on the push condition so that we don't generate a build for every push to main (which is every merged PR).  Limiting builds to those that change the docker files or Cargo.toml could help.

Maybe add a workflow_dispatch condition so that we can ask for the image to be made manually instead.

Reply:
don't we need a fresh image each time we change IPA code because it carries the helper binary?

Reply:
Agree with Alex that we should generate docker image each time helper code changes so we have the latest helper image.

Reply:
We would need a fresh docker image each time only for testing purpose. This can be done manually using workflow dispatch. When in future, we run a scheduled test at a fixed cadence, we can change the cron to generate a new image each time. This should be sufficient for now and we can revisit it later.